win32/tests/job_object_helper_ut/job_object_helper_ut.c

```diff
@@ -0,0 +1,386 @@
+TEST_FUNCTION(job_object_helper_create_succeeds)
+{
+    // arrange
+    setup_job_object_helper_create_expectations();
+
+    // act
+    THANDLE(JOB_OBJECT_HELPER) job_object_helper = job_object_helper_create();
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    // cleanup
+    THANDLE_ASSIGN(JOB_OBJECT_HELPER)(&job_object_helper, NULL);
+
+}
+
+/*Tests_SRS_JOB_OBJECT_HELPER_18_018: [ If there are any failures, job_object_helper_create shall fail and return NULL. ]*/
+TEST_FUNCTION(job_object_helper_create_fails)
+{
+    // arrange
+    setup_job_object_helper_create_expectations();
+    umock_c_negative_tests_snapshot();
+
+    for (size_t i = 0; i < umock_c_negative_tests_call_count(); i++)
+    {
+        if (umock_c_negative_tests_can_call_fail(i))
+        {
+            umock_c_negative_tests_reset();
+            umock_c_negative_tests_fail_call(i);
+
+            // act
+            THANDLE(JOB_OBJECT_HELPER) result = job_object_helper_create();
+
+            // assert
+            ASSERT_IS_NULL(result);
+        }
+    }
+
+}
+
+/*Tests_SRS_JOB_OBJECT_HELPER_18_033: [ job_object_helper_dispose shall call CloseHandle to close the handle to the job object. ]*/
+TEST_FUNCTION(job_object_helper_dispose_succeeds)
+{
+    // arrange
+    setup_job_object_helper_create_expectations();
+    THANDLE(JOB_OBJECT_HELPER) job_object_helper = job_object_helper_create();
+    umock_c_reset_all_calls();
+
+    STRICT_EXPECTED_CALL(interlocked_decrement(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(mocked_CloseHandle(test_job_object));
+    STRICT_EXPECTED_CALL(free(IGNORED_ARG));
+
+    // act
+    THANDLE_ASSIGN(JOB_OBJECT_HELPER)(&job_object_helper, NULL);
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+}
+
+/*Tests_SRS_JOB_OBJECT_HELPER_18_035: [ If job_object_helper is NULL, job_object_helper_limit_memory shall fail and return a non-zero value. ]*/
+TEST_FUNCTION(job_object_helper_limit_memory_with_NULL_job_object_helper)
+{
+    // arrange
+
+    // act
+    int result = job_object_helper_limit_memory(NULL, 42);
+
+    // assert
+    ASSERT_ARE_NOT_EQUAL(int, 0, result);
+
+}
+
+/*Tests_SRS_JOB_OBJECT_HELPER_18_036: [ If percent_physical_memory is 0, job_object_helper_limit_memory shall fail and return a non-zero value. ]*/
+/*Tests_SRS_JOB_OBJECT_HELPER_18_037: [ If percent_physical_memory is greater than 100, job_object_helper_limit_memory shall fail and return a non-zero value. ]*/
+TEST_FUNCTION(job_object_helper_limit_memory_with_invalid_percent_physical_memory)
+{
+    // arrange
+    setup_job_object_helper_create_expectations();
+    THANDLE(JOB_OBJECT_HELPER) job_object_helper = job_object_helper_create();
+    umock_c_reset_all_calls();
+
+    uint32_t invalid_values[] = {0, 101, 143};
+
+    for (int i=0; i < _countof(invalid_values); i++)
```

Comment:
>_countof [](http://example.com/codeflow?start=22&length=8)

suggest to use sizeof(invalid_values)/sizeof(invalid_values[0]) which is portable